### PR TITLE
nrf/mpy: Set features consistent with documentation and other ports.

### PR DESCRIPTION
**Pull request by mzdaniel**
While following https://docs.micropython.org/en/latest/reference/mpyfiles.html,
I stumbled upon sys.implementation.mpy. Turns out MICROPY_PERSISTENT_CODE_LOAD is needed for it and this PR proposes to bring NRF in par with the other archs.

Next step was to follow https://docs.micropython.org/en/latest/develop/natmod.html and again was crossed with compatibility issues. Today, the master branch builds micropython with no unicode as reflected by
```
>>> print('0x%02x' % (sys.implementation.mpy >> 8))
0x00
```
Building the documented factorial example renders an incompatible mpy that has unicode enabled by default, shown in its 3rd byte:
```
$ hexdump -C -n 4 factorial.mpy 
00000000  4d 05 0b 1f      |M...|
```
We dont want to enable by default features that compromize precious flash size, so I tried adding  MPY_CROSS_FLAGS += -mno-unicode  in the Makefile, but the unicode bit was still present after recompilation.
I've been thinking to look deeper here for a PR, and that's when I realized unicode is the default across other archs. At this point, enabling unicode still doesnt make mpy compatible but nonetheless seems much better to be consistent with the other archs.

The final part was to find configs that allows matching feature flags as required by the micropython import system. Long story short, mpy-cross unconditionally emits thumb and inline thumb code and this needs to be reflected in the target arch. No all archs agree here, and looks like the best practice is to enable them conditionally according to stm32. After applying this PR on nrf52840 the data will look like:
```
>>> print('0x%02x' % (sys.implementation.mpy >> 8))
0x1e
```
```
$ hexdump -C -n 4 factorial.mpy 
00000000  4d 05 1e 1f      |M...|
```
In short with this PR, NRF users of nrf9160 and nrf52840 series (1M flash) should be able to follow the docs as intended. For nrf52832 (512K/256K flash) and nrf51822 (256K/128K flash) I am not so sure what defaults are best.

### CODE REVIEW

**Comment by dpgeorge** [comment]
Merged in 8f45f5ee4f58c64226a25a7070de0d18727be273
